digest.go

```diff
@@ -109,18 +109,11 @@ func (d Digest) Validate() error {
 		return ErrDigestInvalidFormat
```

Comment:
> If returning `ErrDigestInvalidFormat` for `foobar:!!!` is seen a more useful error than `ErrDigestUnsupported`, which I am not convinced that either is more actionable than the other…

The distinction is that `ErrDigestUnsupported` is “hmm, I haven't heard of that one”, while `ErrDigestInvalidFormat` is “that is wrong!  Nobody should accept it”.  The latter you can reject in middleman software (e.g. registries) regardless of who the end consumers will eventually be.

> … then only do this regexp match when the algorithm is not recognized to return the right error.

Yeah, I'm fine with that.